index.go

```diff
@@ -142,6 +148,42 @@ func (idx *faissIndex) Search(x []float32, k int64) (
 	return
 }
 
+func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64) (
+	distances []float32, labels []int64, err error,
+) {
+	excludeSelector, err := NewIDSelectorNot(exclude)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var sp *C.FaissSearchParameters
+	C.faiss_SearchParameters_new(&sp, (*C.FaissIDSelector)(excludeSelector.sel))
+	ivfPtr := C.faiss_IndexIVF_cast(idx.cPtr())
+	if ivfPtr != nil {
+		sp = C.faiss_SearchParametersIVF_cast(sp)
+		C.faiss_SearchParametersIVF_new_with_sel(&sp, (*C.FaissIDSelector)(excludeSelector.sel))
+	}
+	defer func() {
```

Comment:
Why defer here, when we can simply call it at the end of the function.
Our practice to use defers should be only when there's multiple exit paths for a method.